src/commands/deployment/README.md

```diff
@@ -25,7 +25,39 @@ datadog-ci deployment mark --env prod --service payment-service --revision v1.1.
 - `--service` the name of the service being deployed. For example, `payment-service`.
 - `--is-rollback` specifies that the deployment is a rollback.
 - `--tags` is an array of key value pairs in the format `key:value`. These tags are added to the deployment event shown in Datadog.
-- `--no-fail` (default: `false`) will prevent the deployment command from failing if there are issues submitting the data.
+- `--no-fail` (default: `false`) prevents the deployment command from failing if there are issues submitting the data.
+
+### `correlate`
+**Note**: If you are using `datadog-ci deployment mark`, then you do not need to use this command, as the correlation is made automatically.
+
+The `correlate` command connects a GitOps deployment with the CI pipeline of the application repository. Once they are connected, you can check which pipeline
+triggered a deployment, and which deployments were triggered by a pipeline.
+
+**Important**: This command does not work for every setup. Refer to [when to call the command](#When-to-call-the-command) for more details.
+
+#### When to call the command
+
+In order for the command to work properly, you first need to commit changes to the configuration repository (where the Kubernetes manifests are) from
+the CI before calling the command:
+
+1. Make the changes to the configuration (for example, updating a image tag).
+2. Run `git commit -m "updating my kubernetes configuration"`.
+3. Run `datadog-ci deployment correlate --provider <cd_provider>` (you can refer to the command syntax below).
+4. Run `git push`.
+
+If you are using [argo cd image updater][3], this command does not work since it relies on making the changes using `git commit`. 
+
+Again, these steps need to happen in your CI since the end goal of this command is to correlate the pipeline doing the configuration changes
+with the associated deployments.
+
+For example:
+```bash
+datadog-ci deployment correlate --provider argocd
+```
+
+- `--provider` (**required**): the CD provider name. Currently, the only supported CD provider is `argocd`.
+- `--config-repo`: configuration repository URL where the kubernetes manifests are stored. If empty, the command tries to get it using a Git command.
```

Comment:
"the command tries to get it using a Git command" seems very general to me. Either we say which command it is, or we change it like "the command tries to extract the repository from the local git files" (or similar)